overlord/ifacestate/udevmonitor/udevmon.go

```diff
@@ -131,21 +157,30 @@ func (m *Monitor) udevEvent(ev *netlink.UEvent) {
 }
 
 func (m *Monitor) addDevice(kobj string, env map[string]string) {
-	di, err := hotplug.NewHotplugDeviceInfo(env)
+	dev, err := hotplug.NewHotplugDeviceInfo(env)
 	if err != nil {
 		return
 	}
+	if _, seen := m.devices[dev.DevicePath()]; seen {
+		return
+	}
+	m.devices[dev.DevicePath()] = true
 	if m.deviceAdded != nil {
-		m.deviceAdded(di)
+		m.deviceAdded(dev)
 	}
 }
 
 func (m *Monitor) removeDevice(kobj string, env map[string]string) {
-	di, err := hotplug.NewHotplugDeviceInfo(env)
+	dev, err := hotplug.NewHotplugDeviceInfo(env)
 	if err != nil {
 		return
 	}
+	if _, seen := m.devices[dev.DevicePath()]; !seen {
```

Comment:
This can also be simplified to just:

```
if !m.devices[dev.DevicePath()] {
...
}
```

This also suggests that the variable should be just called `seen` rather than `devices`